gdb-server/src/writer.rs

```diff
@@ -1,58 +1,57 @@
 use async_std::{io::Write, net::TcpStream, prelude::*};
 use futures::channel::mpsc;
 use gdb_protocol::packet::{CheckedPacket, Kind as PacketKind};
-use std::sync::Arc;
 
 type Result<T> = std::result::Result<T, Box<dyn std::error::Error + Send + Sync>>;
 type Sender<T> = mpsc::UnboundedSender<T>;
 
 pub async fn writer(
     packet: CheckedPacket,
-    stream: Arc<TcpStream>,
-    packet_stream: Sender<CheckedPacket>,
+    stream: &mut TcpStream,
+    packet_stream: &Sender<CheckedPacket>,
     buffer: &mut Vec<u8>,
 ) -> Result<()> {
     let mut tmp_buf = [0; 128];
     log::debug!("WRITE WIN");
-    if packet.is_valid() {
```

Comment:
Why did you remove this check? Did you put it some other place? If yes, I cannot find it :/

Reply:
As I understand, `CheckedPacket` is guaranteed to be valid. 
Is is created from an`UncheckedPacket` by checking if it is valid, so we don't have to check again here.